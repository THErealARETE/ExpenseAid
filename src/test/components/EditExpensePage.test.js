import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
//import moment from 'moment'
import Adapter from 'enzyme-adapter-react-16';
import EditExpensePage from '../../components/EditExpensePage'; 
import expense from '../fixtures/expenses';
Enzyme.configure({
    adapter: new Adapter()
})

let editExpense ,removeExpense,  history, wrapper 

beforeEach(()=>{
     editExpense = jest.fn()
     removeExpense = jest.fn()
     history = {push: jest.fn()}
     wrapper = shallow(<EditExpensePage
         editExpense={editExpense}
         removeExpense= {removeExpense}
          history={history}
          expense = {expense[2]}
          />)
})
test('should render edit expense page correctly',()=>{
    
    expect(wrapper).toMatchSnapshot()
})

// should hadle edit expense correctly
test('should handle onSubmit correctly',()=>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expense[2])
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(editExpense).toHaveBeenLastCalledWith(expense[2].id, expense[2])
})


//should handle remove expense correctly 
test('should handle removeExpense correctly',()=>{
    wrapper.find('button').simulate('click')
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(removeExpense).toHaveBeenLastCalledWith(
        {id:expense[2].id})
})