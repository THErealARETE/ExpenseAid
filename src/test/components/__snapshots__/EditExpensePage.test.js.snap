// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render edit expense page correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(EditExpensePage)
    editExpense={[MockFunction]}
    expense={
      Object {
        "amount": 400,
        "createdAt": 345600000,
        "description": "arm",
        "id": "3",
        "note": "",
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    removeExpense={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
